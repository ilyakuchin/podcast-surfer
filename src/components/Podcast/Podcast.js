import React, { Component } from "react";
import { Link } from "react-router-dom";
import axios from "axios";

export default class Podcast extends Component {
  constructor(props) {
    super(props);

    this.state = { episodes: [] };
  }
  render() {
    return this.state ? (
      <div>
        <div>{this.state.name}</div>
        <div>{this.state.description}</div>
        <img
          width="200"
          height="200"
          src={this.state.image}
          alt="podcast cover"
        />
        <ul>
          {this.state.episodes.map(item => (
            <li key={item.id}>
              <Link
                to={{
                  pathname: "/episode-player",
                  state: {
                    name: item.name,
                    description: item.description,
                    image: item.image,
                    audio: item.audio
                  }
                }}
              >
                {item.name}
              </Link>
              <div>{item.description}</div>
              <img
                width="200"
                height="200"
                src={item.image}
                alt="episode cover"
              />
            </li>
          ))}
        </ul>
      </div>
    ) : null;
  }

  componentDidMount() {
    const { rss } = this.props.location.state;
    axios
      .get(`${process.env.REACT_APP_API_URL}/podcast?rss=${rss}`)
      .then(res => {
        this.setState({ ...res.data });
      });
  }
}
