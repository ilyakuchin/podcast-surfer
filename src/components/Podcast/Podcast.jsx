import React from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import LoadingSpinner from '../LoadingSpinner/LoadingSpinner';
import ConnectedEpisodesList from '../EpisodesList/EpisodesList';

const PodcastGrid = styled.div`
  display: grid;
  width: 100%
  margin-top: 30px;
  justify-items: center;
  grid-gap: 15px;
  grid-template-columns: auto;
  grid-template-rows: auto;
  grid-template-areas:
    "podcast-img podcast-title"
    "podcast-img podcast-description"
    "episodes episodes";

  @media(max-width: 600px) {
    grid-template-areas:
    "podcast-img"
    "podcast-title"
    "podcast-description"
    "episodes";
  }
`;

const PodcastImg = styled.img`
  grid-area: podcast-img;
  margin-left: 10px;
  width: 300px;
  height: 300px;
  object-fit: scale-down;

  @media (max-width: 600px) {
    margin: 0;
    max-width: 65%;
    max-height: 65%;
  }
`;

const PodcastTitle = styled.h2`
  grid-area: podcast-title;
  margin: 0;
  margin-left: 10px;
  margin-right: 10px;
  @media (max-width: 600px) {
    text-align: center;
  }
`;

const PodcastDescription = styled.p`
  grid-area: podcast-description;
  max-width: 700px;
  white-space: pre-line;

  @media (max-width: 600px) {
    margin-left: 10px;
    margin-right: 10px;
  }
`;

export function Podcast({ isFetching, name, description, imageUrl }) {
  return !isFetching ? (
    <PodcastGrid>
      <PodcastImg src={imageUrl} alt='podcast cover' />
      <PodcastTitle>{name}</PodcastTitle>
      <PodcastDescription>{description}</PodcastDescription>
      <ConnectedEpisodesList />
    </PodcastGrid>
  ) : (
    <LoadingSpinner />
  );
}

Podcast.propTypes = {
  isFetching: PropTypes.bool.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired
};

const mapStateToProps = state => {
  return {
    isFetching: state.currentPodcast.isFetching,
    name: state.currentPodcast.name,
    description: state.currentPodcast.description,
    imageUrl: state.currentPodcast.imageUrl
  };
};

const ConnectedPodcast = connect(mapStateToProps)(Podcast);

export default ConnectedPodcast;
