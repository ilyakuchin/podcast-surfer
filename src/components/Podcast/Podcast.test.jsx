import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Podcast } from './Podcast';

configure({ adapter: new Adapter() });

describe('Test Podcast Component', () => {
  test('should render correctly', () => {
    const wrapper = shallow(
      <Podcast
        name='podacstName'
        description='podcastDescription'
        imageUrl='podcast image'
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
