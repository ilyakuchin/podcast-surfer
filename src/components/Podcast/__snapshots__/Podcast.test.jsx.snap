// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Podcast Component should render correctly 1`] = `
<styled.div>
  <styled.img
    alt="podcast cover"
    src="podcast image"
  />
  <styled.h2>
    podacstName
    <Connect(SubscribeButton) />
  </styled.h2>
  <styled.p>
    podcastDescription
  </styled.p>
  <Connect(EpisodesList) />
</styled.div>
`;

exports[`Test Podcast Component should render correctly when isFetching 1`] = `
<styled.div>
  <LoadingSpinner />
</styled.div>
`;
